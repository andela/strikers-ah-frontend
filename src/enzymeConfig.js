import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-localstorage-mock';

export default Enzyme.configure({
  adapter: new Adapter(),
});
