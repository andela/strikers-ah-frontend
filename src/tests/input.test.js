/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import Input from '../components/common/input';
import '../enzymeConfig';

const props = {
  labelStyle: 'label-styles',
  name: 'firstname',
  type: 'text',
  inputStyle: 'input-style',
  label: 'label',
  innerLabelStyle: 'inner-style',
  error: 'firstname must be 2 length in character',
  borderStyle: 'border-style',
  errorStyle: 'error-style',
  onChange: jest.fn(),
};

describe('<Input/>', () => {
  it('Should render <Input/> without crashing', () => {
    const input = shallow(<Input {...props} />);
    expect(input).toMatchSnapshot();
  });
});
