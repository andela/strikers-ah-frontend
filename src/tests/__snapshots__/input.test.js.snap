// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input/> Should render <Input/> without crashing 1`] = `ShallowWrapper {}`;
